### PR TITLE
Fixed #7 issue

### LINKED ISSUE #7: Possible bug in `bounced<TokenTransferInternal>` receiver
Duplicate `self.balance`reference:

https://github.com/tact-lang/jetton/blob/543803a2929929bc7fe2c9d93e3454745bebd401/sources/jetton_wallet.tact#L139

```tact
bounced(msg: bounced<TokenTransferInternal>){
    self.balance += self.balance + msg.amount;
}
```

Pointed out by Alexey Malanov.

### PR DESCRIPTION
**Pull request by Shvandre**
Closes #7 

Typo, but may be a security breach. (Though those messages should never be bounced, but maybe it is possible to make them bounce back)

### CODE REVIEW

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Please add regression tests

File: sources/contract.spec.ts [hunk 1]
```diff
@@ -529,7 +529,129 @@ describe("JettonMinter", () => {
 
         expect(await deployerJettonWallet.getJettonBalance()).toEqual(initialJettonBalance);
     });
+    describe('Bounces', () => {
+        // This is borrowed from stablecoin, and is not implemented here.
+        // Should it be implemented?
```
Comment (anton-trunov) [inline sources/contract.spec.ts:534]: why is it a comment here and not an open issue?

File: sources/contract.spec.ts [hunk 2]
```diff
@@ -529,7 +529,129 @@ describe("JettonMinter", () => {
...
+        it('minter should restore supply on internal_transfer bounce', async () => {
+            const deployerJettonWallet    = await userWallet(deployer.address);
+            const mintAmount = BigInt(getRandomInt(1000, 2000));
+            const mintMsg = beginCell().store(storeMint({$$type: "Mint", amount: mintAmount, receiver: deployer.address})).endCell();
+            //const mintMsg    = JettonMinter.mintMessage(jettonMinter.address, deployer.address, mintAmount,toNano('0.1'), toNano('0.1'));
```
Comment (anton-trunov) [inline sources/contract.spec.ts:539]: there should be no commented out code; if you need it for some reason, please explain it as a comment

File: sources/contract.spec.ts [hunk 3]
```diff
@@ -529,7 +529,129 @@ describe("JettonMinter", () => {
...
+            const transferMsg = beginCell().store(storeTokenTransfer({$$type: "TokenTransfer",
+                query_id: 0n,
+                amount: txAmount,
+                response_destination: deployer.address,
+                destination: notDeployer.address,
+                custom_payload: null,
+                forward_ton_amount: 0n,
+                forward_payload: beginCell().endCell().beginParse()
+            })).endCell()
+            //const transferMsg = JettonWallet.transferMessage(txAmount, notDeployer.address, deployer.address, null, 0n, null);
```
Comment (anton-trunov) [inline sources/contract.spec.ts:590]: same here

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Please open an issue about un-testability of the fix we discussed elsewhere

File: sources/contract.spec.ts (hunk 1 above)
Comment (Shvandre) [reply sources/contract.spec.ts:534]: Should I remove it?

Comment (anton-trunov) [reply sources/contract.spec.ts:534]: it's better to understand what it means and try to deal with it

**Comment by Shvandre** [comment]
Opened an issue and mentioned it in the code comment. @anton-trunov Can you, please, review this, so I will rebase GasTests on new main?

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**